### PR TITLE
[FIX] xmlrpc: properly marshall date & datetime objects

### PR DESCRIPTION
**Pull request by Elkasitu**
Before this revision, trying to read a Date field using xmlrpc would
result in a marshalling error.

This is due to the fact that 960360afe478a8f7b9c456721b5591154952a37d
changed the internal value of Date/Datetime fields to hold
datetime.date/datetime objects.

The xmlrpc layer is capable of converting datetimes to the XML
datetime representation, but this is not possible for datetime.date
objects.

In order to solve this, the Marshaller class used by python's
xmlrpc.client is overridden and monkey-patched, allowing to convert
datetime.date / datetime.datetime objects into properly formatted
strings.

opw-1896364

### CODE REVIEW

**Comment by Elkasitu** [comment]
robodoo r+

**Comment by Elkasitu** [comment]
no u

**Comment by rco-odoo** [comment]
robodoo please r+

File: odoo/addons/base/controllers/rpc.py [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+    dispatch[datetime] = dump_datetime
+
+    def dump_date(self, value, write):
+        value = Date.to_string(value)
+        self.dump_unicode(value, write)
+    dispatch[date] = dump_date
+
+
+# monkey-patch xmlrpc.client's marshaller
+xmlrpc.client.Marshaller = OdooMarshaller
```
Comment (xmo-odoo) [inline odoo/addons/base/controllers/rpc.py]: Why monkey-patch the XML-RPC marshaller instead of just using the custom marshaller to dump in _xmlrpc?

**Comment by rco-odoo** [comment]
robodoo r-

**Comment by Elkasitu** [comment]
robodoo get back to work!!

**Comment by rco-odoo** [comment]
robodoo review+ again

**Comment by EchPay** [comment]
Hi, I can't understand why we have to overwrite a python lib. The case is when I try to call xml-rpc from Odoo to other system, It can occur falure because of wrong format datetime

**Pull request closed without merging**